docker/nginx/conf.d/server/server.api

```diff
@@ -137,15 +137,14 @@ location /abuse/report {
 location /hns {
     include /etc/nginx/conf.d/include/cors;
 
-    # match the request_uri and extract the hns domain and anything that is passed in the uri after it
-    # example: /hns/something/foo/bar matches:
-    # > hns_domain: something
-    # > path: /foo/bar/
-    set_by_lua_block $hns_domain { return string.match(ngx.var.uri, "/hns/([^/?]+)") }
-    set_by_lua_block $path { return string.match(ngx.var.uri, "/hns/[^/?]+(.*)") }
-
-    proxy_set_header Host $host;
-    include /etc/nginx/conf.d/include/location-hns;
+    rewrite_by_lua_block {
+        local hns_domain = string.match(ngx.var.uri, "/hns/([^/?]+)")
+        local path = string.match(ngx.var.uri, "/hns/[^/?]+(.*)")
+        local args = ngx.var.args and ngx.var.is_args .. ngx.var.args or ""
```

Comment:
Q: I'm struggling to understand this line. What is the difference between `args` and `is_args`? What is the order of operations here? I figure `..` is the concatenation operator but what are we concatenating?

Reply:
so `is_args` is the query params connector

> $is_args
“?” if a request line has arguments, or an empty string otherwise

would it be clearer if we do sth like this?

```lua
local args = ngx.var.args and "?" .. ngx.var.args or ""
```

Reply:
Okay, that looks right I think. I'm still confused about the order of operations, in my head it's:

```
(ngx.var.args and ngx.var.is_args) .. (ngx.var.args or "")
```

but I'm not sure if that's right.

Reply:
> Okay, that looks right I think. I'm still confused about the order of operations, in my head it's:
> 
> ```
> (ngx.var.args and ngx.var.is_args) .. (ngx.var.args or "")
> ```
> 
> but I'm not sure if that's right.

`..` is not a logical operation so it's evaluated first, it's

```lua
(ngx.var.args and (ngx.var.is_args .. ngx.var.args)) or ""
```

Reply:
That makes sense. Wish that was more clear, but it is what it is.

Reply:
> That makes sense. Wish that was more clear, but it is what it is.

I think since that coding style in lua raised questions several times already, I might make a full `if` out of it next time or take some time to make these already existing ones more readable.